src/sandstorm/sandstorm-http-bridge.capnp

```diff
@@ -41,3 +41,47 @@ interface SandstormHttpBridge {
   # If BridgeConfig.saveIdentityCaps is true for this app, adds the given identity to the
   # grain's database, allowing it to be fetched later with `getSavedIdentity()`.
 }
+
+interface AppHooks (AppObjectId) {
+  # When connecting to the bridge's domain socket at /tmp/sandstorm-api, the
+  # application may supply this as a bootstrap interface. If the app chooses
+  # to do so, it should also set `bridgeConfig.expectAppHooks = true` in the
+  # package's PackageDefinition.
+
+  getViewInfo @0 () -> Grain.UiView.ViewInfo;
+  # Like Grain.UiView.GetViewInfo. If AppHooks is supplied, the bridge will
+  # delegate UiView.GetViewInfo to this method. If it raises unimplemented,
+  # the bridge will fall back to reading the viewInfo from the bridgeConfig.
+
+  restore @1 (objectId :AppObjectId) -> (cap :Capability);
+  # Like Grain.MainView.restore. If an app implements AppHooks.restore, any
+  # capabilities it provides which implement AppPersistent should return
+  # a value of type BridgeObjectId (below), with the `application` variant
+  # set to the value that should be passed to AppHooks.restore/drop.
+
+  drop @2 (objectId :AppObjectId);
+  # Like Grain.MainView.drop. See the comments for restore.
+
+  struct BridgeObjectId {
```

Comment:
I think this struct doesn't need to be public at all if sandstorm-http-bridge automatically wraps/unwraps `AppObjectId` into/out of the private version of `BridgeObjectId`.

Reply:
See my other comment, but I'm unsure how to do the wrapping part, since that happens in AppPersistent.save()